modules/custom/d_content/modules/d_content_init/src/Services/ConfigUpdate.php

```diff
@@ -41,65 +42,138 @@ public function __construct(ConfigFactory $configFactory, ModuleHandler $moduleH
 
   /**
    * Searches for all files in the given path that contain in the name the
-   * value specified in the variable $configName. Then, in the configurations
-   * found, it changes the value of the theme option to the default theme.
+   * value specified in the variable $configName.
    *
    * @param string $moduleName
    *   Module name.
    * @param string $path
-   *   Directory path with configs.
+   *   The path to the directory with the configuration for the module.
    * @param string $regex
    *   Regular expresion pattern to search in configuration file names.
    *
    * @return array
-   *   Array of configs.
+   *   Array of configs names.
    */
-  public function getConfigs($moduleName, $path, $regex) {
-    $configs = [];
-    if ($this->moduleHandler->moduleExists($moduleName)) {
-      $dir = $this->moduleHandler->getModule($moduleName)->getPath();
-      if (file_exists($dir)) {
-        $files = scandir($dir . $path);
-        foreach ($files as $file) {
-          if (preg_match($regex, $file) === 1) {
-            $configs[] = substr($file, 0, -4);
-          }
-        }
+  public function getConfigsFilesName($moduleName, $path, $regex) {
+    $configsFileNames = [];
+    $filesNames = scandir($this->getConfigsPath($moduleName, $path));
```

Comment:
mi osobiscie nie bardzo sie podoba ten scandir( [direcotry] ) - to [direcotry] nie koniecznie musi byc zwrocone przez getConfigsPath prawda? jak moduł nie istnieje, to zwroci null i beda leciały warningi `If directory is not a directory, then boolean FALSE is returned, and an error of level E_WARNING is generated`. Moze zapiszmy to do zmiennej i wykonujmy kod dalej jezeli zmianna bedzie miala jakas wartosc?

Reply:
Jeżeli nie ma katalogu to jest rzucany wyjątek. Tutaj brakło try catch :)